_test/outputxmlhelper.test.php

```diff
@@ -83,4 +92,71 @@ public function parameterProviderForXMLCallWithFloatParams()
             'invalid start = 1.1 and count = 02.00' => [1.1, 02.00]
         ];
     }
+
+    /**
+     * Test to ensure that the getUrlParam method gets the correct url params set in the url
+     * and formats them as integers, and not as strings.
+     */
+    public function test_geturlparam_returns_set_url_parameters()
+    {
+        $startName = 'start';
+        $countName = 'count';
+        $defaultStart = 0;
+        $defaultCount = 20;
+
+        $expectedStartValue = 1;
+        $expectedCountValue = 5000;
+
+        $getParamWithRandomParams = ['asd' => 'hehe', 'demo' => 'eeeeh', 'start' => '1', 'count' => '5000'];
```

Comment:
Use `$expectedStartValue` and `$expectedCountValue` instead of `1` and `5000`.

Reply:
Also, good use of decoy data 👍

Reply:
Yes I could use them, but I would need to convert them to a string, because the superglobal `$_GET` only has strings and therefor I expect a string in my `\OutputXMLHelper::getUrlParam` method.  

I also think that it makes no sense to have an integer, that I convert to a string, and afterwards is converted to an integer again. 

We could

 1. → convert them to a string when giving them to the array fragment.
 2. → not convert them and provide them as integer.
 3. → leave it like that.

Please choose wise :wink:

Reply:
I choose option A ... I mean 1.

Reply:
`"$expectedStartValue"` should work fine btw